src/Symfony/Component/AssetMapper/ImportMap/ImportMapGenerator.php

```diff
@@ -232,10 +237,11 @@ private function findEagerImports(MappedAsset $asset): array
                 // should not happen at this point, unless something added a bogus JavaScriptImport to this asset
                 throw new LogicException(sprintf('Cannot find imported JavaScript asset "%s" in asset mapper.', $javaScriptImport->assetLogicalPath));
             }
+
             $dependencies = array_merge($dependencies, $this->findEagerImports($nextAsset));
```

Comment:
I wouldn't use array_merge in the loop https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop

Reply:
If this is something we try to avoid in other parts of Symfony, then let's do it (but I'm not sure it is). In this case, this code is called in dev but not in prod (the result is cached in prod). So it's not SUCH an important spot for performance.

Reply:
Sometimes with each new SF version dev env feels slower and slower..

Reply:
@norkunas do you suggest a foreach.. or to just use the array_unique at the end ?

Reply:
Rewrote the traversal .. low memory + better handling of duplicates